### PR TITLE
[Finder] update finder comparator to php 7.1

### PR DESCRIPTION
**Pull request by rvitaliy**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes if this classes was extended
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | no
| License       | MIT
| Doc PR        | no

### CODE REVIEW

File: src/Symfony/Component/Finder/Comparator/DateComparator.php [hunk 1]
```diff
@@ -9,6 +9,8 @@
  * file that was distributed with this source code.
  */
 
+declare(strict_types=1);
```
Comment (javiereguiluz) [inline src/Symfony/Component/Finder/Comparator/DateComparator.php]: I guess this change should be reverted. There's an ongoing discussion about adding it or not to new code (see #28423).

Comment (rvitaliy) [reply src/Symfony/Component/Finder/Comparator/DateComparator.php]: waiting for approval of RFC ❤️

**Comment by nicolas-grekas** [comment]
Thanks @rvitaliy for submitting this PR. I'm going to close it because merging it would break Symfony's BC policy that we all rely on to ensure a safe migration path to the latest versions. Here are some details about it if you want to read more about it: https://symfony.com/bc

**Comment by rvitaliy** [comment]
@Nicals any add of typehinting to existing class is BC. 
What could be the solution to this problem? 😕

**Comment by nicolas-grekas** [comment]
It's not really a problem: everything works well also without. The good news is that php 7.2 will allow us to do that in Symfony 5, in one year.

**Pull request closed without merging**